File: api/src/main/java/dev/ulloasp/mlsuite/team/application/dto/TeamPermissionsDto.java
package dev.ulloasp.mlsuite.team.application.dto;

public record TeamPermissionsDto(
        boolean canViewTeam,
        boolean canEditTeam,
        boolean canDeleteTeam,
        boolean canViewTeamMembers,
        boolean canManageTeamMemberRoles,
        boolean canRemoveTeamMembers) {
}

File: frontend/src/api/workspace/dtos/team-permissions-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

export interface TeamPermissionsDto {
  canViewTeam: boolean;
  canEditTeam: boolean;
  canDeleteTeam: boolean;
  canViewTeamMembers: boolean;
  canManageTeamMemberRoles: boolean;
  canRemoveTeamMembers: boolean;
}
